module/DTO/questionsDTO.js

```diff
@@ -0,0 +1,577 @@
+                    : "";
+              obj[childrenKey] = slider;
+            }
+          }
+          childQuestion[key] = obj;
+        } else if (key === "payload") {
+          // Assign payload data
+          childQuestion[key] = {
+            ...keyData,
+            criteriaId: childData.identifier,
+            responseType: type,
+          };
+        } else if (key === "file") {
+          // Handle file-related data
+          if (
+            childData[keyData["type"]] &&
+            childData[keyData["type"]].length > 0
+          ) {
+            const require = childData?.interactions?.validation?.required;
+            childQuestion[key] = {
+              ...keyData,
+              required:
+                require === true || require === "Yes" || require === "yes"
+                  ? "Yes"
+                  : "No",
+              type: childData?.evidence?.type,
+            };
+          } else childQuestion[key] = "";
+        } else if (key === "updatedAt" || key === "createdAt") {
+          // Assign timestamps
+          childQuestion[key] = child[keyData];
+        } else if (key === "dateFormat") {
+          // Assign date format
+          childQuestion[key] =
+            childData?.interactions?.response1?.validation?.pattern || "";
+        } else if (key === "options") {
+          // Assign options
+          childQuestion[key] =
+            childData?.interactions?.response1?.options || [];
+        } else if (key === "showRemarks" || key === "autoCapture") {
+          // Handle boolean fields
+          const typeofChildren = childData[keyData];
+          childQuestion[key] =
+            key !== "autoCapture"
+              ? typeofChildren === true ||
+                typeofChildren === "Yes" ||
+                typeofChildren === "yes"
+                ? true
+                : false
+              : childData[keyData] || false;
+        } else if (key === "questionNumber" || key === "page") {
+          // Assign question number and page
+          childQuestion[key] =
+            key === "questionNumber" ? `${index + 1}` : baseAssessment.page;
+        } else {
+          // Assign other fields
+          childQuestion[key] = childData[keyData]
+            ? childData[keyData]
+            : child[keyData]
+              ? child[keyData]
+              : "";
+        }
+      }
+      // Resolve with the transformed question data
+      resolve({
+        success: true,
+        message: messageConstants.apiResponses.EVIDENCE_FETCHED,
+        data: {
+          ...childQuestion,
+        },
+      });
+    });
+  }
+
+  /**
+   * Generates matrix questions based on the criteria.
+   * @method
+   * @name getMatrixQuestions
+   * @param {Object} criteria - The criteria object.
+   * @returns {Object} - The matrix questions object.
+   */
+  static getMatrixQuestions(criteria) {
+    const matrixObj = {};
+    for (let key in questionType.matrix) {
+      if (key === "instanceIdentifier") {
```

Comment:
@jayesh9747 plesae add all strings in common file. If it is relevent to migration then migration common.js else in the service common.js do this for all the files